src/main.py

```diff
@@ -1,196 +1,211 @@
+async def on_guild_available(guild: discord.Guild):
+    if not dbg.is_debug_bot():
+        import context
+        await client.setup_guild(guild)
+
+    await tr.refresh_db(guild)
+
+    # This is 100% going to cause issues if we ever want to host on more than one server
+    # TODO: If we want to fix this, make announcement channels a list in config.json, and add a server ID column to DB
+    game_channel = discord.utils.get(guild.text_channels, id=GAME_ANNOUNCEMENT_CHANNEL)
+
+    if game_channel is not None:
+        print(f"Announcing games in server '{guild.name}' channel '{game_channel.name}'")
+    else:
+        await client.close()
+        raise Exception(f"Game announcement error: couldn't find channel {GAME_ANNOUNCEMENT_CHANNEL}")
+
+    game_timer.start(game_channel)
+    thermo.start(guild)
+
+    # Set Bouncer's status
+    await update_user_count(guild)
+
+"""
+On Member Join
+
+Runs when a user joins the server
+"""
+@client.event
+async def on_member_join(user: discord.Member):
+    await update_user_count(user.guild)
+
+"""
+On Member Remove
+
+Runs when a member leaves the server
+"""
+@client.event
+async def on_member_remove(user: discord.Member):
+    tr.remove_from_cache(user.id)
+    await update_user_count(user.guild)
+
+"""
+On Message
+
+Runs when a user posts a message
+"""
+@client.event
+async def on_message(message: discord.Message):
+    # Ignore bots completely (including ourself)
+    if message.author.bot:
+        return
+
+    # For now, completely ignore DMs
+    if isinstance(message.channel, discord.channel.DMChannel):
+        return
+
+    # Check first if we're toggling debug mode
+    # Need to do this before we discard a message
+    if dbg.check_toggle(message):
+        await dbg.toggle_debug(message)
+        return
+    elif dbg.should_ignore_message(message):
+        return
+
+    # Keep track of the user's message for dynamic slowmode
+    await thermo.user_spoke(message)
+    # Check if we need to congratulate a user on getting a new role
+    # Don't award XP if posting in specified disabled channels
+    if message.channel.id not in XP_OFF:
+        lvl_up_message = await tr.give_xp(message.author, message.guild)
+        if lvl_up_message:
+            await message.channel.send(lvl_up_message)
+
+    for log_link in re.findall(r"https://smapi.io/log/[a-zA-Z0-9]{32}", message.content):
+        log_dict = requests.get(f"http://api.pil.ninja/smapi_log/endpoint?{log_link}").json()
+        if log_dict["success"]:
+            windows_info = re.search(r"(Windows (?:Vista|\d+)) .+", log_dict["OS"])
+            if windows_info:  # Condense OS text for Windows because it's often quite verbose.
+                log_dict["OS"] = windows_info.group(1)
+            await message.channel.send(smapi_log_message_template.substitute(log_dict))
+
+    for attachment in message.attachments:
+            if attachment.filename == "SMAPI-latest.txt":
+                print("SMAPI-latest.txt found")

```

Comment:
idk if we need to print this out to the console, it just clutters things up, remove.